Confidence/src/main/java/com/spotify/confidence/AndroidLifecycleEventProducer.kt

```diff
@@ -128,10 +128,10 @@ class AndroidLifecycleEventProducer(
         val currentVersion = ConfidenceValue.String(packageInfo?.versionName ?: "")
         val currentBuild = ConfidenceValue.String(packageInfo?.getVersionCode().toString() ?: "")
 
-        // Get the previous recorded version.
-        val previousVersion = sharedPreferences
-            .getString(APP_VERSION, null)
-            ?.let(ConfidenceValue::String)
+        contextFlow.value =
+            mapOf(APP_VERSION_KEY to currentVersion)
+        contextFlow.value =
+            mapOf(APP_BUILD_KEY to currentBuild)
```

Comment:
contextFlow is of type `MutableStateFlow` that seems to lose some updates if we change value too quickly. I noticed app_build being set to the context, but not app_version.
I think it's better to fix this in this PR

Reply:
oh yes. the way we set the context flow here is wrong. mutableStateFlow is designed to only keep the last value. which is what we want, however in setting it, we need to change the whole value. in this case, we need to get the current value, which is a map, update the map entry and put back the whole map again into the contextFlow value. we can create a function to do all that and use it every where. the helper function can accept a key value to be updated in the value of context flow map.